packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,193 @@
+import { Namespaces } from 'src/i18n'
+import {
+  convertDollarsToMaxSupportedPrecision,
+  convertLocalAmountToDollars,
+} from 'src/localCurrency/convert'
+import { useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { getLocalCurrencyExchangeRate, getLocalCurrencySymbol } from 'src/localCurrency/selectors'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { getRecentPayments } from 'src/send/selectors'
+import { isPaymentLimitReached, showLimitReachedError } from 'src/send/utils'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { dollarBalanceSelector } from 'src/stableToken/selectors'
+
+const { decimalSeparator } = getNumberFormatSettings()
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeAmount>
+
+type Props = RouteProps
+
+const oneDollarAmount = {
+  value: new BigNumber('1'),
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+export function ExchangeTradeScreen({ navigation, route }: Props) {
+  function isNextButtonValid() {
+    const passMinAmount = dollarAmount.isGreaterThanOrEqualTo(
+      isAddFunds ? DOLLAR_ADD_FUNDS_MIN_AMOUNT : DOLLAR_CASH_OUT_MIN_AMOUNT
+    )
+    const passMaxAmount =
+      isAddFunds || (dollarBalance && dollarAmount.isLessThanOrEqualTo(dollarBalance))
+
+    return dollarAmount && passMinAmount && passMaxAmount
+  }
+
+  function onChangeExchangeAmount(amount: string) {
+    setInputAmount(amount)
+  }
+
+  function goNext() {
+    const now = Date.now()
+    const isLimitReached = isPaymentLimitReached(now, recentPayments, dollarAmount.toNumber())
+    if (isLimitReached) {
+      dispatch(showLimitReachedError(now, recentPayments, localExchangeRate, localCurrencySymbol))
+      return
+    }
+
+    navigation.navigate(Screens.FiatExchangeOptions, {
+      amount: dollarAmount,
+      currencyCode: localCurrencyCode,
+      isAddFunds,
+    })
+  }
+
+  const { isAddFunds } = route.params
+  const { t } = useTranslation(Namespaces.fiatExchangeFlow)
+  const dispatch = useDispatch()
+  const [inputAmount, setInputAmount] = React.useState('')
+  const dollarBalance = useSelector(dollarBalanceSelector)
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const recentPayments = useSelector(getRecentPayments)
+
+  const parsedInputAmount = parseInputAmount(inputAmount, decimalSeparator)
+  const dollarAmount = convertDollarsToMaxSupportedPrecision(
+    (!parsedInputAmount.isNaN() &&
+      convertLocalAmountToDollars(parsedInputAmount, localCurrencyCode ? localExchangeRate : 1)) ||
+      new BigNumber('0')
+  )
+
+  return (
+    <SafeAreaView
+      // Force inset as this screen uses auto focus and KeyboardSpacer padding is initially
+      // incorrect because of that
+      forceInset={{ top: 'never', bottom: 'always' }}
```

Comment:
Nit: pull this object out so we don't remake it each time? Probably doesn't matter but its what we do with other styles